ocp_resources/metallb.py

```diff
@@ -0,0 +1,59 @@
+# Reference Doc: https://github.com/metallb/metallb-operator/blob/main/api/v1beta1/metallb_types.go
+
+from ocp_resources.resource import NamespacedResource
+
+
+class MetalLB(NamespacedResource):
+    """
+    MetalLB object.
+    """
+
+    api_group = NamespacedResource.ApiGroup.METALLB_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        log_level=None,
+        speaker_config=None,
+        speaker_tolerations=None,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the MetalLB or it's CR's.
+            namespace (str): Namespace of the MetalLB
+            client: (DynamicClient): DynamicClient to use.
+            log_level(str): The verbosity of the controller and the speaker logging.
+                            Allowed values are: all, debug, info, warn, error, none. (default: info)
+            speaker_config (dict): Additional configs to be applied on MetalLB Speaker daemonset.
+            speaker_tolerations (list): List of tolerations to be applied on Speaker Daemonset.
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            **kwargs,
+        )
+        self.log_level = log_level
+        self.speaker_config = speaker_config
+        self.speaker_tolerations = speaker_tolerations
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res["spec"] = {}
+            available_log_levels = ["all", "debug", "info", "warn", "error", "none"]
+            if self.log_level and self.log_level in available_log_levels:
+                self.res["spec"]["logLevel"] = self.log_level
+            else:
```

Comment:
> I will also not check if it in the accepted list, the list may be changed and we do not want to keep track on it.

I took this list from here --> https://github.com/metallb/metallb-operator/blob/main/api/v1beta1/metallb_types.go#L29-L37

Reply:
> `self.logger.info( f"Given input doesn't matched {available_log_levels}.Default 'log_level' is set to 'info'" )`
> 
> you ignore the user request and set info, we need to raise if the user sent us unsupported level

I set logger.info because of the if any other string is passed then it ignores it and set `log_level` to `info`. 
I am also mentioning the supported/available log_levels in the message.

Reply:
> > `self.logger.info( f"Given input doesn't matched {available_log_levels}.Default 'log_level' is set to 'info'" )`
> > you ignore the user request and set info, we need to raise if the user sent us unsupported level
> 
> I set logger.info because of the if any other string is passed then it ignores it and set `log_level` to `info`. I am also mentioning the supported/available log_levels in the message.

https://github.com/metallb/metallb-operator/blob/main/api/v1beta1/metallb_types.go#L56-L60

Reply:
you can't always monitor the upstream changes.
we can set INFO as default if not set, we do not correct the user, we need to pass the user data

Reply:
Updated.